### PR TITLE
Issue #3085248: Replaced individual ::load calls in loop with ::loadMultiple along with some code adjustments.

### PR DESCRIPTION
**Pull request by navneet0693**
## Problem
Avoid calls to ::load() in loops.

## Solution
1. Use ::loadMultiple
2. Refactor the code to put most logical if conditions to be checked first.

## Issue tracker
https://www.drupal.org/project/social/issues/3085248

## How to test
- N.A

## Release notes
N.A

## Change Record
N.A

### CODE REVIEW

File: modules/custom/social_path_manager/social_path_manager.module [hunk 1]
```diff
@@ -102,13 +102,15 @@ function social_path_manager_batch_alter(&$batch) {
...
+ *
+ * @throws \Exception
  */
 function _social_path_manager_update_group_tab_aliases($id, $action) {
   $group_ids = array_keys(\Drupal::entityQuery('group')->execute());
   $created = 0;
 
-  foreach ($group_ids as $group_id) {
-    $group = Group::load($group_id);
+  $groups = Group::load($group_ids);
```
Comment (bramtenhove) [inline modules/custom/social_path_manager/social_path_manager.module]: @navneet0693 shouldn’t this be Group::loadMultiple()?

**Comment by jaapjan** [comment]
@navneet0693 I restarted the test. Hope it helps. 

Could you also reply on @bramtenhove comment?

**Comment by navneet0693** [comment]
> @navneet0693 I restarted the test. Hope it helps.
> 
> Could you also reply on @bramtenhove comment?

Resolved.

**Comment by jaapjan** [comment]
I think this still needs work considering the test is failing:
https://travis-ci.org/goalgorilla/open_social/jobs/592570876

**Pull request merged**